modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala

```diff
@@ -136,24 +148,76 @@ abstract class BspTestDefinitions(val scalaVersionOpt: Option[String])
     )
     inputs.fromRoot { root =>
       os.proc(TestUtil.cli, "setup-ide", ".", extraOptions).call(cwd = root, stdout = os.Inherit)
-      val bspFile = root / ".bsp" / "scala-cli.json"
-      expect(os.isFile(bspFile))
-      val json = ujson.read(
-        os.read(bspFile: os.ReadablePath, charSet = Codec(Charset.defaultCharset()))
+      readBspConfig(
+        root,
+        details => {
+          expect(details.argv.length >= 2)
+          expect(details.argv(1) == "bsp")
+        }
       )
-      // check that we can decode the connection details
-      val details = upickle.default.read(json)(detailsCodec)
-      expect(details.argv.length >= 2)
-      expect(details.argv(1) == "bsp")
     }
   }
 
+  for (command <- Seq("setup-ide", "compile", "run"))
+    test(command + " should result in generated bsp file") {
+      val inputs = TestInputs(
+        Seq(
+          os.rel / "simple.sc" ->
+            s"""val msg = "Hello"
+               |println(msg)
+               |""".stripMargin
+        )
+      )
+      inputs.fromRoot { root =>
+        os.proc(TestUtil.cli, command, ".", extraOptions).call(cwd = root, stdout = os.Inherit)
+        readBspConfig(
+          root,
+          details => {
+            expect(details.argv.length >= 2)
+            expect(details.argv(1) == "bsp")
+            expect(details.argv(3) == "--json-options")
+            expect(Paths.get(details.argv(4)).isAbsolute)
+          }
+        )
+
+        val scalaCliBspConfigFile = root / ".scala" / "ide-options.json"
+        expect(scalaCliBspConfigFile.toIO.exists())
```

Comment:
NIT os-lib can do that:
```suggestion
        expect(os.exists(scalaCliBspConfigFile))
```